clients/roscpp/include/ros/service_server_link.h

```diff
@@ -106,6 +106,7 @@ class ROSCPP_DECL ServiceServerLink : public boost::enable_shared_from_this<Serv
    * it has finished.
    */
   bool call(const SerializedMessage& req, SerializedMessage& resp);
+  bool call(const SerializedMessage& req, SerializedMessage& resp, double timeout);
```

Comment:
@paulbovbel I haven't changed `double timeout` to `ros::Duration` or `ros::WallDuration` because internally it's used with `boost::chrono::steady_clock` (as suggested by @flixr , instead of `boost::chrono::system_clock`).

Reply:
It could still be a `ros::WallDuration` as you can just call `toSec()` or `toNSec` on it.
Personally I don't have a strong opinion on whether to use a simple double or WallDuration here.

Reply:
Changed to `ros::WallDuration`.